config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.hypertrace.config.service;
+
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.grpc.stub.StreamObserver;
+import java.util.LinkedHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.config.service.store.ConfigStore;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.GetConfigRequest;
+import org.hypertrace.config.service.v1.GetConfigResponse;
+import org.hypertrace.config.service.v1.UpsertConfigRequest;
+import org.hypertrace.config.service.v1.UpsertConfigResponse;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+public class ConfigServiceGrpcImpl extends ConfigServiceGrpc.ConfigServiceImplBase {
+
+  private static final String DEFAULT_CONTEXT = "DEFAULT-CONTEXT";
+
+  private final ConfigStore configStore;
+
+  public ConfigServiceGrpcImpl(ConfigStore configStore) {
+    this.configStore = configStore;
+  }
+
+  @Override
+  public void upsertConfig(UpsertConfigRequest request,
+      StreamObserver<UpsertConfigResponse> responseObserver) {
+    try {
+      UpsertConfigResponse response = configStore
+          .writeConfig(getConfigResource(request), getUserId(), request.getConfig());
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Upsert failed for request:{}", request, e);
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void getConfig(GetConfigRequest request,
+      StreamObserver<GetConfigResponse> responseObserver) {
+    try {
+      GetConfigResponse response;
+      if (request.getConfigVersion() != 0) {  // if configVersion is present
+        String context = null;
+        // get the config version corresponding to the most specific context(last element in the contexts list)
+        if (!request.getContextsList().isEmpty()) {
+          context = request.getContexts(request.getContextsCount() - 1);
```

Comment:
This is an interesting question with unclear behavior - if I ask for a version and a list of scopes, we're only returning the last. That's unexpected to me, and I think indicates that the version doesn't belong in the same api. My suggestion would be to remove version from the get API and when/if needed, introduce a new API for a versioned request that does not support merging/multiple contexts.

Reply:
Removed version from `getConfig` API but still keeping it in store, so that we can easily support the new API(for getting a specific version of config) easily.